### DISCUSSION TITLE
Any way to use countUp to update a property/class on an HTML element?

### DISCUSSION THREAD

**Discussion by barryculhane** [inorganik/countUp.js#328] [General]
I'm using Bootstrap to render a progress bar in an Angular app.  I use countUp to _animate_ a counter value above the progress bar.

It would be nice to be able to animate the progress bar's value at the same time as the counter value is being counted up.

Is there any way to use countUp in an Angular component's code to update one of the component's property values?

_countUp_ the counterValue component property  to animate the progress bar ?
`<div class="progress-bar" attr.style="width:{{counterValue}}%"></div>`

**Comment by inorganik** [comment]
Sorry for the late reply (2 years 😬) but yes, you can use the `frameVal` property in the progress bar. Its the unformatted number on any given frame.
```html
<div class="progress-bar" attr.style="width:{{countUp.frameVal}}%"></div>
```
Just a note about using countUp in angular, for ngx-countUp we run outside of zone, because otherwise each frame will trigger a change detection cycle. See: https://github.com/inorganik/ngx-countUp/blob/master/projects/ngx-countup/src/lib/ngx-countup.directive.ts#L74

If you set up your animation like the countUp directive linked above, it should work.
